tests/components/vodafone_station/test_config_flow.py

```diff
@@ -228,3 +228,75 @@ async def test_options_flow(
     assert result["data"] == {
         CONF_CONSIDER_HOME: 37,
     }
+
+
+async def test_reconfigure_successful(
+    hass: HomeAssistant,
+    mock_vodafone_station_router: AsyncMock,
+    mock_setup_entry: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test that the host can be reconfigured."""
+    mock_config_entry.add_to_hass(hass)
+    result = await mock_config_entry.start_reconfigure_flow(hass)
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "reconfigure"
+
+    # original entry
+    assert mock_config_entry.data["host"] == "fake_host"
+
+    reconfigure_result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            "host": "192.168.100.60",
+            "password": "fake_password",
+            "username": "fake_username",
+        },
+    )
+    await hass.async_block_till_done()
```

Comment:
I don't think you need this. Did you find that it was needed?

```suggestion
    await hass.async_block_till_done()
```

Reply:
Actually there is a good mix of places with and without this call after `hass.config_entries.flow.async_configure()`

Anyway, removed for this PR